src/cartservice/src/Program.cs

```diff
@@ -42,29 +42,30 @@
 
 builder.Services.AddSingleton<ICartStore>(cartStore);
 
-builder.Services.AddOpenTelemetryTracing((builder) => builder
-    .ConfigureResource(r => r
-        .AddTelemetrySdk()
-        .AddEnvironmentVariableDetector()
-        .AddDetector(new DockerResourceDetector())
-    )
-    .AddRedisInstrumentation(
-        cartStore.GetConnection(),
-        options => options.SetVerboseDatabaseStatements = true)
-    .AddAspNetCoreInstrumentation()
-    .AddGrpcClientInstrumentation()
-    .AddHttpClientInstrumentation()
-    .AddOtlpExporter());
+// see https://opentelemetry.io/docs/instrumentation/net/getting-started/
 
-builder.Services.AddOpenTelemetryMetrics(builder => builder
-    .ConfigureResource(r => r
-        .AddTelemetrySdk()
-        .AddEnvironmentVariableDetector()
-        .AddDetector(new DockerResourceDetector())
-    )
-    .AddRuntimeInstrumentation()
-    .AddAspNetCoreInstrumentation()
-    .AddOtlpExporter());
+var appResourceBuilder = ResourceBuilder
+    .CreateDefault()
+    .AddTelemetrySdk()
+    .AddEnvironmentVariableDetector()
+    .AddDetector(new DockerResourceDetector());
+
+builder.Services.AddOpenTelemetry()
+    .WithTracing(builder => builder
+        .SetResourceBuilder(appResourceBuilder)
+        .AddRedisInstrumentation(
+            cartStore.GetConnection(),
+            options => options.SetVerboseDatabaseStatements = true)
+        .AddAspNetCoreInstrumentation()
+        .AddGrpcClientInstrumentation()
+        .AddHttpClientInstrumentation()
+        .AddConsoleExporter()
```

Comment:
Please remove the `ConsoleExporter`.

With this, the docker compose logs are bloated with cart-service entries:

```
cart-service             | GetCartAsync called with userId=
cart-service             | Activity.TraceId:            947e807dda97f7e45514843779a68bec
cart-service             | Activity.SpanId:             ddee9a80987ed0f0
cart-service             | Activity.TraceFlags:         Recorded
cart-service             | Activity.ParentSpanId:       455744d34f754ee7
cart-service             | Activity.ActivitySourceName: OpenTelemetry.Instrumentation.AspNetCore
cart-service             | Activity.DisplayName:        oteldemo.CartService/GetCart
cart-service             | Activity.Kind:               Server
cart-service             | Activity.StartTime:          2023-02-20T06:18:36.3621416Z
cart-service             | Activity.Duration:           00:00:00.0011445
cart-service             | Activity.Tags:
cart-service             |     net.host.name: cartservice
cart-service             |     net.host.port: 7070
cart-service             |     http.method: POST
cart-service             |     http.scheme: http
cart-service             |     http.target: /oteldemo.CartService/GetCart
cart-service             |     http.url: http://cartservice:7070/oteldemo.CartService/GetCart
cart-service             |     http.flavor: 2.0
cart-service             |     http.user_agent: grpc-node-js/1.6.7
cart-service             |     app.user.id:
cart-service             |     app.cart.items.count: 0
cart-service             |     http.status_code: 200
cart-service             |     rpc.system: grpc
cart-service             |     net.peer.ip: ::ffff:172.19.0.22
cart-service             |     net.peer.port: 49548
cart-service             |     rpc.service: oteldemo.CartService
cart-service             |     rpc.method: GetCart
cart-service             |     rpc.grpc.status_code: 0
cart-service             | Activity.Events:
cart-service             |     Fetch cart [02/20/2023 06:18:36 +00:00]
cart-service             | Resource associated with Activity:
cart-service             |     container.id: 81633081208a8d00bb50f85fa36ed32e43c459a8c10ca92abbf7532357c1c18d
cart-service             |     service.name: cartservice
cart-service             |     service.namespace: opentelemetry-demo
cart-service             |     telemetry.sdk.name: opentelemetry
cart-service             |     telemetry.sdk.language: dotnet
cart-service             |     telemetry.sdk.version: 1.4.0.788
```

Reply:
done